zetaclient/context/chain.go

```diff
@@ -0,0 +1,166 @@
+package context
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/pkg/errors"
+	"golang.org/x/exp/maps"
+	"golang.org/x/exp/slices"
+
+	"github.com/zeta-chain/zetacore/pkg/chains"
+	observer "github.com/zeta-chain/zetacore/x/observer/types"
+)
+
+// ChainRegistry is a registry of supported chains
+type ChainRegistry struct {
+	chains map[int64]Chain
+
+	// additionalChains is a list of additional static chain information to use when searching from
+	// chain IDs. It's stored in the protocol to dynamically support new chains without doing an upgrade
+	additionalChains []chains.Chain
+
+	mu sync.Mutex
+}
+
+// Chain represents chain with its parameters
+type Chain struct {
+	id       int64
+	chain    *chains.Chain
+	params   *observer.ChainParams
```

Comment:
I would call the two field
- chainInfo
- observerParams
To make more explicit

Reply:
https://github.com/zeta-chain/node/pull/2568/commits/2d6e3123e996155b088619478a74076bc14af928